### PR TITLE
feat: store multiple tokens

### PR DESCRIPTION
**Pull request by Prakhar-Agarwal-byte**
Token management for #3273 

- Update PersistantState schema to have a list of saved tokens for multiple accounts
- Create migrations for the same

### CODE REVIEW

**Comment by sandipndev** [comment]
@Prakhar-Agarwal-byte unit tests are failing, it expects the new schema. Can you fix that?

File: app/store/persistent-state/state-migrations.ts [hunk 1]
```diff
@@ -30,9 +30,24 @@ type PersistentState_6 = {
   galoyAuthToken: string
 }
 
-const migrate6ToCurrent = (state: PersistentState_6): Promise<PersistentState> =>
+type PersistentState_7 = {
+  schemaVersion: 7
+  galoyInstance: GaloyInstanceInput
+  galoyAuthToken: string
+  galoySavedAccounts: string[]
```
Comment (sandipndev) [inline app/store/persistent-state/state-migrations.ts:37]: Perhaps we can be a little more explicit?

```suggestion
  galoyAuthToken: string
  galoyOtherAuthTokens: string[]
```

File: app/hooks/use-app-config.ts [hunk 2]
```diff
@@ -35,6 +40,7 @@ export const useAppConfig = () => {
           return {
             ...state,
             galoyAuthToken: token,
+            galoySavedAccounts: [...state.galoySavedAccounts, token],
```
Comment (sandipndev) [inline app/hooks/use-app-config.ts:43]: Should also update the `saveTokenAndInstance` function.

**Review by sandipndev — COMMENTED** [review]
Good initial draft, I'll need to review this PR locally once after the comments are addressed and the tests are passing before approval.

File: app/store/persistent-state/state-migrations.ts (hunk 1 above)
Comment (Prakhar-Agarwal-byte) [reply app/store/persistent-state/state-migrations.ts:37]: `galoySavedAccounts` will contain the current token as well so maybe we can rename it to `galoyAllAuthTokens` instead?

File: app/hooks/use-app-config.ts (hunk 2 above)
Comment (Prakhar-Agarwal-byte) [reply app/hooks/use-app-config.ts:43]: Sure

**Comment by nicolasburtey** [comment]
I think this PR should also be merged first: https://github.com/GaloyMoney/blink-mobile/pull/3122

Before we think of multiple token, we need to make storing tokens safer

File: app/store/persistent-state/state-migrations.ts [hunk 3]
```diff
@@ -30,9 +30,24 @@ type PersistentState_6 = {
...
+}
+
+const migrate7ToCurrent = (state: PersistentState_7): Promise<PersistentState> =>
   Promise.resolve(state)
 
+const migrate6ToCurrent = (state: PersistentState_6): Promise<PersistentState> => {
+  return migrate7ToCurrent({
+    ...state,
+    schemaVersion: 7,
+    galoyAllAuthTokens: [],
```
Comment (sandipndev) [inline app/store/persistent-state/state-migrations.ts:47]: Shouldn't this check if the user has a token preset and add it in the All array?

Comment (Prakhar-Agarwal-byte) [reply app/store/persistent-state/state-migrations.ts:47]: Yes it should be present in the all tokens list

**Comment by sandipndev** [comment]
I will do a thorough testing on logged in phones / non logged in phones and verify the user is not logged out before merging.

**Comment by sandipndev** [comment]
Verified and merged!

**Pull request merged**